array.go

```diff
@@ -1193,3 +1193,143 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// IsOrdered succeeds if array is ordered based on optional `comparator` function.
+// For default, it will use built-in comparator function for each data type.
+// It requires all elements in the array to have same data type
```

Comment:
Let's clarify this a bit:

`It requires all elements in the array to have same data type` => `Built-in comparator requires all elements in the array to have same data type.`